pkgs/stdenv/generic/make-derivation.nix

```diff
@@ -314,7 +315,9 @@ else let
           else "${attrs.pname}${staticMarker}${hostSuffix}-${attrs.version}"
         );
     }) // lib.optionalAttrs __structuredAttrs { env = checkedEnv; } // {
+      # NOTE: keep in sync with trivial-builders.nix#inputDerivation
       builder = attrs.realBuilder or stdenv.shell;
+      # NOTE: keep in sync with trivial-builders.nix#inputDerivation
```

Comment:
It was disproportionally difficult to share a function to DRY this. `make-derivation.nix` isn't set up to have access to a lot of things. I've resolved it be deleting the extra function, because there's no immediate need and `inputDerivation` has an uncertain future anyway.